pkg/apis/resource/validation/validation.go

```diff
@@ -83,11 +84,29 @@ func ValidateResourceClaimUpdate(resourceClaim, oldClaim *resource.ResourceClaim
 	return allErrs
 }
 
+// ResourceClaimStatusValidationOptions contains the different settings for ResourceClaim validation
+type ResourceClaimStatusValidationOptions struct {
+	// Allow invalid label-value in NodeSelector
+	AllowInvalidLabelValueInNodeSelector bool
+}
+
+func ValidationOptionsForResourceClaimStatus(resourceClaim, oldClaim *resource.ResourceClaim) ResourceClaimStatusValidationOptions {
+	opts := ResourceClaimStatusValidationOptions{
+		AllowInvalidLabelValueInNodeSelector: false,
+	}
+	if oldClaim != nil && oldClaim.Status.Allocation != nil &&
+		oldClaim.Status.Allocation.NodeSelector != nil {
+		terms := oldClaim.Status.Allocation.NodeSelector.NodeSelectorTerms
+		opts.AllowInvalidLabelValueInNodeSelector = helper.HasInvalidLabelValueInNodeSelectorTerms(terms)
```

Comment:
if at all possible, I'd like to see this be unconditionally disallowed as part of promotion to beta